lib/Transforms/ConvertSecretInsertToStaticInsert/ConvertSecretInsertToStaticInsert.cpp

```diff
@@ -0,0 +1,146 @@
+
+#include <utility>
+
+#include "lib/Analysis/SecretnessAnalysis/SecretnessAnalysis.h"
+#include "llvm/include/llvm/ADT/STLExtras.h"    // from @llvm-project
+#include "llvm/include/llvm/Support/Casting.h"  // from @llvm-project
+#include "mlir/include/mlir/Analysis/DataFlow/ConstantPropagationAnalysis.h"  // from @llvm-project
+#include "mlir/include/mlir/Analysis/DataFlow/DeadCodeAnalysis.h"  // from @llvm-project
+#include "mlir/include/mlir/Analysis/DataFlowFramework.h"  // from @llvm-project
+#include "mlir/include/mlir/Dialect/Affine/IR/AffineOps.h"  // from @llvm-project
+#include "mlir/include/mlir/Dialect/Arith/IR/Arith.h"    // from @llvm-project
+#include "mlir/include/mlir/Dialect/SCF/IR/SCF.h"        // from @llvm-project
+#include "mlir/include/mlir/Dialect/Tensor/IR/Tensor.h"  // from @llvm-project
+#include "mlir/include/mlir/IR/Diagnostics.h"            // from @llvm-project
+#include "mlir/include/mlir/IR/ImplicitLocOpBuilder.h"   // from @llvm-project
+#include "mlir/include/mlir/IR/MLIRContext.h"            // from @llvm-project
+#include "mlir/include/mlir/IR/Operation.h"              // from @llvm-project
+#include "mlir/include/mlir/IR/PatternMatch.h"           // from @llvm-project
+#include "mlir/include/mlir/IR/Value.h"                  // from @llvm-project
+#include "mlir/include/mlir/Interfaces/SideEffectInterfaces.h"  // from @llvm-project
+#include "mlir/include/mlir/Support/LLVM.h"           // from @llvm-project
+#include "mlir/include/mlir/Support/LogicalResult.h"  // from @llvm-project
+#include "mlir/include/mlir/Transforms/GreedyPatternRewriteDriver.h"  // from @llvm-project
+
+namespace mlir {
+namespace heir {
+
+#define GEN_PASS_DEF_CONVERTSECRETINSERTTOSTATICINSERT
+#include "lib/Transforms/ConvertSecretInsertToStaticInsert/ConvertSecretInsertToStaticInsert.h.inc"
+
+struct SecretInsertToStaticInsertConversion
+    : OpRewritePattern<tensor::InsertOp> {
+  using OpRewritePattern<tensor::InsertOp>::OpRewritePattern;
+
+ public:
+  SecretInsertToStaticInsertConversion(DataFlowSolver *solver,
+                                       MLIRContext *context)
+      : OpRewritePattern(context), solver(solver) {}
+
+  LogicalResult matchAndRewrite(tensor::InsertOp insertOp,
+                                PatternRewriter &rewriter) const override {
+    // TODO(#866): Add support for writes on multi-dimensional tensors
+    auto index = insertOp->getOperand(2);
+    auto tensor = insertOp->getOperand(1);
+    auto insertedValue = insertOp->getOperand(0);
+
+    auto *indexSecretnessLattice =
+        solver->lookupState<SecretnessLattice>(index);
+
+    // Implies that the tensor.insert op is newly created by this pass
+    if (!indexSecretnessLattice) {
+      insertOp->emitWarning()
+          << "Secretness for tensor.insert index has not been set";
+      return failure();
+    }
+
+    bool isIndexSecret = indexSecretnessLattice->getValue().getSecretness();
```

Comment:
I think this might still cause crashes, because you're checking only for the lattice element being null (so the `->` is safe) but not for unitialzied, so the `.getSecretness` might crash with an assert